core/src/test/java/com/linecorp/armeria/client/endpoint/SlowStartAwareEndpointSelectionStrategyTest.java

```diff
@@ -0,0 +1,139 @@
+
+        EndpointGroupRegistry.register(groupName, endpointGroup, strategy);
+
+        assertThat(EndpointGroupRegistry.selectNode(ctx, groupName).authority()).isEqualTo("127.0.0.1:1234");
+        assertThat(EndpointGroupRegistry.selectNode(ctx, groupName).authority()).isEqualTo("127.0.0.1:2345");
+        assertThat(EndpointGroupRegistry.selectNode(ctx, groupName).authority()).isEqualTo("127.0.0.1:3456");
+        assertThat(EndpointGroupRegistry.selectNode(ctx, groupName).authority()).isEqualTo("127.0.0.1:1234");
+        assertThat(EndpointGroupRegistry.selectNode(ctx, groupName).authority()).isEqualTo("127.0.0.1:2345");
+        assertThat(EndpointGroupRegistry.selectNode(ctx, groupName).authority()).isEqualTo("127.0.0.1:3456");
+    }
+
+    @Test
+    void weightedRoundRobin() {
+        final EndpointGroup endpointGroup = EndpointGroup.of(
+                Endpoint.of("127.0.0.1", 1234).withWeight(1),
+                Endpoint.of("127.0.0.1", 2345).withWeight(2),
+                Endpoint.of("127.0.0.1", 3456).withWeight(3));
+        final String groupName = "weighted";
+
+        EndpointGroupRegistry.register(groupName, endpointGroup, strategy);
+
+        assertThat(EndpointGroupRegistry.selectNode(ctx, groupName).authority()).isEqualTo("127.0.0.1:1234");
+        assertThat(EndpointGroupRegistry.selectNode(ctx, groupName).authority()).isEqualTo("127.0.0.1:2345");
+        assertThat(EndpointGroupRegistry.selectNode(ctx, groupName).authority()).isEqualTo("127.0.0.1:3456");
+        assertThat(EndpointGroupRegistry.selectNode(ctx, groupName).authority()).isEqualTo("127.0.0.1:2345");
+        assertThat(EndpointGroupRegistry.selectNode(ctx, groupName).authority()).isEqualTo("127.0.0.1:3456");
+        assertThat(EndpointGroupRegistry.selectNode(ctx, groupName).authority()).isEqualTo("127.0.0.1:3456");
+    }
+
+    @Test
+    void selectFromDynamicEndpointGroup() throws Exception {
+        final AtomicInteger counter = new AtomicInteger();
+        final EndpointWeightTransition weighter = (e, step, maxStep) -> {
+            counter.incrementAndGet();
+            return EndpointWeightTransition.linear().compute(e, step, maxStep);
+        };
+        final EndpointSelectionStrategy strategy =
+                SlowStartAwareEndpointSelectionStrategy.of(weighter,
+                                                           scheduler,
+                                                           Duration.ofSeconds(1),
+                                                           10);
+
+        final TestDynamicEndpointGroup endpointGroup = new TestDynamicEndpointGroup();
+        EndpointGroupRegistry.register("dynamic", endpointGroup, strategy);
+        endpointGroup.updateEndpoints(ImmutableList.of(Endpoint.of("127.0.0.1", 1000)));
+
+        final EndpointSelector selector = EndpointGroupRegistry.getNodeSelector("dynamic");
+        // Schedule updateEndpointWeight.
+        TimeUnit.SECONDS.sleep(1);
```

Comment:
It might be good to have a test that uses a real executor to make sure it is used, but I think making the `updateWeight` method `@VisibleForTesting` and testing a complex series of transitions without `await()` and sleeps would be easier to reason about.